ext/curl/curl.stub.php

```diff
@@ -3553,12 +3553,80 @@
  */
 const CURLOPT_SAFE_UPLOAD = UNKNOWN;
 
+/**
+ * @strict-properties
+ */
+class CurlException extends \Exception {};
+
+/**
+ * @strict-properties
+ */
+class CurlHandleException extends \CurlException {};
+
+/**
+ * @strict-properties
+ */
+class CurlMultiException extends \CurlException {};
+
+/**
+ * @strict-properties
+ */
+class CurlShareException extends \CurlException {};
+
 /**
  * @strict-properties
  * @not-serializable
  */
 final class CurlHandle
 {
+	public function __construct(?string $uri = null) {}
+
+	/** @alias curl_errno */
+	public function errno(): int {}
+
+	/** @alias curl_error */
+	public function error(): string {}
+
+	/** @alias curl_strerror */
+	static public function strerror(int $code): string {}
+
+	/** @alias curl_escape */
+	public function escape(string $str): string {}
+
+	/** @alias curl_unescape */
+	public function unescape(string $str): string {}
+
+	/**
+	 * If CURLOPT_RETURNTRANSFER is True, returns the result on success,
+	 * otherwise returns True, and the result will be output.
+	 * @throws CurlHandleException on failure.
+	 */
+	public function exec(): string|true {}
+
+	/** @alias curl_pause */
+	public function pause(int $flags): int {}
+
+	/** @alias curl_reset */
+	public function reset(): void {}
+
+	/** @alias curl_getinfo */
+	public function getInfo(?int $option = null): mixed {}
+
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+	/** @alias curl_upkeep */
+	public function upkeep(): bool {}
+#endif
+
+	/**
+	 * Varies from curl_setopt() to allow fluent chaining.
+	 * @throws CurlHandleException
+	 */
+	public function setOpt(int $option, mixed $value): \CurlHandle {}
+	/**
+	 * Returns self to match setOpt() behavior.
+	 * @throws CurlHandleException
+	 */
+	public function setOptArray(array $option): \CurlHandle {}
```

Comment:
```suggestion
	public function setOptArray(array $options): \CurlHandle {}
```
Same as in curl_setopt_array